### ISSUE TITLE
AppX identityName / MakePRI Problem

### ISSUE THREAD

**Issue by markusmoenig** [electron-userland/electron-builder#3023]
Version 20.15.1 / Windows X64
Electron 2.02

The identityName of my application is "CompanyName.Material-Z", the - seems to invoke an error in MakePRI.exe which quits with "Appx manifest not found or is invalid".

If I change the name to "CompanyName.MaterialZ" it works, but this is not the identity name I use in the Windows AppStore.... . As far as I know Material-Z should be valid and the AppStore did not complain about it when creating the identity.

Any chance to get "Material-Z" to work ?

**Comment by markusmoenig** [comment]
@black-snow @develar ? Can't submit my app with this ...

**Comment by black-snow** [comment]
If makepri doesn't like `-` in `identityName` *imo* you should
* check that you have the latest SDK
* reach out to MS and ask them for advice

**Comment by markusmoenig** [comment]
@black-snow I have the latest SDK installed. 

Somehow I doubt that this is an MS issue as there are apps with "-" inside the Windows AppStore.

**Comment by black-snow** [comment]
Please test the exe directly. If it does not accept your name there's nothing electron-builder can do about it. If it does then we must be messing something up.

According to [this ](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/appxmanifestschema/element-identity)it should work: "A string between 3 and 50 characters in length that consists of alpha-numeric, period, and dash characters."

**Comment by markusmoenig** [comment]
MakePRI does not have problems with files with "-" in the filename I think, I think the manifest file is not in the expected location. With "MaterialZ" it seems to be there.

**Comment by markusmoenig** [comment]
@black-snow Anything else I could try ? Just want to get the app into the store ...

**Comment by black-snow** [comment]
I, sadly, do not have time to investigate this myself.

You could follow the build steps and try to figure out what goes wrong. You can alway clone the repo and modify the (non-binary) files for debugging.

Then there is the MS support which was great the last time I contacted them.

And we can summon @develar upon this Issue 👺 - if he has time ;)
He's always busy with this huge project but probably also the one to fix this one in a blink of an eye.

**Comment by markusmoenig** [comment]
Just trying to keep this alive ... @black-snow @develar

**Comment by black-snow** [comment]
You are a dev, right? Why don't you debug it yourself? Or did you hit a dead end? Let me (us) know what you find (found).

**Comment by markusmoenig** [comment]
@black-snow Yes I am a developer but no idea about Windows and only have a small Windows laptop for compiling / testing. Tried to figure out what is going on but gave up after 1-2 hours.

**Comment by develar** [comment]
I have a huge backlog of issues and issues are donated / from large companies, sorry, I will try to check next week. Have you tried to ask apps store support?

**Comment by markusmoenig** [comment]
I can imagine you are swamped @develar so I appreciate your reply.

I send an eMail to my apps store support contact but he was on vacation and just returned and I guess he is also busy with the backlog now because he did not reply yet.

**Comment by markusmoenig** [comment]
@black-snow @develar I just heard back from my app store contact. Appearantly a dash is not valid in the package id. It means I have to use "MaterialZ" as a product name in the store but later there is a trick to change the display name to "Material-Z" again.

Sorry for driving everybody crazy about this. I will close this for now and let's see how it goes.

Thanks.

**Comment by develar** [comment]
MS is still MS? What a shocking news... 

Validation will be added. Tool must do it, but... hi, It is MS...

**Comment by develar** [comment]
Ok, WTF as always. Waiting until all this crazy *** will be not cleared up. Since it is MS, we cannot do anything without direct support from MS.

**Comment by markusmoenig** [comment]
@black-snow I will ask ... .

**Comment by markusmoenig** [comment]
@black-snow @develar 
The Material-Z is now in the Windows AppStore, submitted it as MaterialZ and later on changed the display name to Material-Z. 

What was a bit strange is that the app got assigned a number for the idenityName in the AppStore (i.e. "CompanyName.Number"), however as a number is not supported for this in any of the tools (makepri) I had to use MaterialZ as the identityName and upon failure correct the generated appx manifest by hand and invoke makeappx.exe manually to build the package. I thought all of this was a bit strange.

The support engineer is trying to find out who is right about the "-" in the identityName (makepri or the  documentation).

**Comment by black-snow** [comment]
@markusmoenig I'm glad you got it to work. So this is an issue with MS and not with the builder (@develar yep, you called this). I'd say we can close the issue.
However, please keep us posted! In case the MS docs are wrong we'll have to change the builder in order to match the new docs (they'll hopefully publish).

Have a great week!
(also cool app you got there!)

**Comment by markusmoenig** [comment]
Thanks @black-snow! Will keep you guys up to date if I hear anything.
